zcutil/release-notes.py

```diff
@@ -0,0 +1,63 @@
+import re, sys, os, os.path
+import subprocess
+import argparse
+from itertools import islice
+
+# Returns dict of {'author': #_of_commits} from a release note
+def authors_in_release_notes(filename):
+    with open(dir + filename, 'r') as f:
+        authors = {}
+        line = f.readline()
+        commits = re.search('(\d+)', line).group(1)
+        first = re.sub(' \(\d+\)|:|\n', '', line)
+        authors[first] = commits
+        for line in f:
+            if line in ['\n', '\r\n']:
+                for author in islice(f, 1):
+                    commit_search = re.search('(\d+)', author)
+                    if commit_search:
+                        commits = commit_search.group(1)
+                    else:
+                        commits = 0
+                    author = re.sub(' \(\d+\)|:|\n', '', author)
+                    authors[author] = commits
+        return authors
+
+## Sums commits made by contributors in each Zcash release note in ./doc/release-notes and writes to authors.md
+def document_authors():
+    print "Writing contributors documented in release-notes directory to authors.md..."
+    with open('../doc/authors.md', 'w') as f:
+        f.write('Zcash Contributors\n==================\n\n')
+        total_contrib = {}
+        for notes in os.listdir(dir):
+            authors = authors_in_release_notes(notes)
+            for author in authors:
+                commits = int(authors[author])
+                if author in total_contrib:
```

Comment:
We should add a mapping before this that links the same author to their various reflog names:
```py
author_aliases = {
    'Simon': 'Simon Liu',
    'bitcartel': 'Simon Liu',
}
```

etc. and then use it here:
```py
for a in authors:
    commits = int(authors[a])
    if a in author_alias:
        author = author_aliases[a]
    else:
        author = a
```

We could do something similar (or just use this itself with e.g. `'Jack Grigg': 'Jack Grigg (str4d)'`) for handling GitHub names, since there's nothing in the reflog that can really help us there.